### PR TITLE
I2c for tizen

### PR DESCRIPTION
**Pull request by tadziopazur**
I2C rework:
 - decouple platform data from generic I2C request handling
 - move platform data into platform component
 - add Tizen I2C port (Tizen API used, NOT linux device APIs)

### CODE REVIEW

**Comment by haesik** [comment]
You should delete Piotr's commit in your PR

File: src/platform/tizen/iotjs_module_i2c-tizen.c [hunk 1]
```diff
@@ -0,0 +1,131 @@
+/* The MIT License (MIT)
```
Comment (LaszloLango) [inline src/platform/tizen/iotjs_module_i2c-tizen.c]: I think we must keep it.

> * The above copyright notice and this permission notice shall be included
> * in all copies or substantial portions of the Software.

Comment (hs0225) [inline src/platform/tizen/iotjs_module_i2c-tizen.c]: @LaszloLango `iotjs_module_i2c-tizen.c`does not have any code referenced from this license. This is a license for `iotjs_module_i2c-linux.c`.

Comment (LaszloLango) [inline src/platform/tizen/iotjs_module_i2c-tizen.c]: I see.

Comment (tadziopazur) [inline src/platform/tizen/iotjs_module_i2c-tizen.c]: Done.

File: src/platform/tizen/iotjs_module_i2c-tizen.c [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  } else if (_this->address != address) {
+    // TODO: report error OR recreate the object with new slave address
+  }
+}
+
+void OpenWorker(uv_work_t* work_req) {
+  I2C_WORKER_INIT_TEMPLATE;
+  // Not used for the deferred construction.
+  // iotjs_i2c_t* i2c = iotjs_i2c_instance_from_reqwrap(req_wrap);
+  // IOTJS_VALIDATED_STRUCT_METHOD(iotjs_i2c_t, i2c);
```
Comment (tadziopazur) [inline src/platform/tizen/iotjs_module_i2c-tizen.c]: Disagree. The deferred initialization is not usually done in iotjs HW components, so when it is done, at least some explanation should be in place.

File: src/modules/iotjs_module_i2c.h [hunk 3]
```diff
@@ -61,12 +57,19 @@ typedef struct {
...
   uint8_t addr;
 #elif defined(__NUTTX__)
+  int device;
   struct i2c_master_s* i2c_master;
   struct i2c_config_s config;
+#elif defined(__TIZEN__)
+  int bus;
+  int address;
+  peripheral_i2c_h handle;
+// For SetAddress call, to reject address changes.
```
Comment (tadziopazur) [inline src/modules/iotjs_module_i2c.h]: Done.

File: src/platform/tizen/iotjs_module_i2c-tizen.c [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+#include <fcntl.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
```
Comment (tadziopazur) [inline src/platform/tizen/iotjs_module_i2c-tizen.c]: Done.

**Comment by tadziopazur** [comment]
PTAL.

**Comment by yichoi** [comment]
LGTM. please check and test again with real target

**Pull request merged**